### DISCUSSION TITLE
K3S Upgrade

### DISCUSSION THREAD

**Discussion by kevinebanks** [k3s-io/k3s#9764] [Q&A]
Hi All,
i am trying to upgrade from  v1.28.7+k3s1 to v1.29.3-k3s1 i have system-upgrade ns and upgrade.yaml below but when i kubectl apply - upgrade.yaml i receive the following error please can someone help.
 
![image](https://github.com/k3s-io/k3s/assets/149391336/f81c7c5e-88fe-4ebb-9140-b358bcf445c2)

![system-upgrade](https://github.com/k3s-io/k3s/assets/149391336/345f71c0-fd6e-4551-ace3-1af85bde1bf6)
  # Server plan
apiVersion: upgrade.cattle.io/v1
kind: Plan
metadata:
  name: server-plan
  namespace: system-upgrade
spec:
  concurrency: 1
  cordon: true
  nodeSelector:
    matchExpressions:
    - key: node-role.kubernetes.io/control-plane
      operator: In
      values:
      - "true"
  serviceAccountName: system-upgrade
  upgrade:
    image: rancher/k3s-upgrade
  version: v1.29.3-k3s1 
---
# Agent plan
apiVersion: upgrade.cattle.io/v1
kind: Plan
metadata:
  name: agent-plan
  namespace: system-upgrade
spec:
  concurrency: 1
  cordon: true
  nodeSelector:
    matchExpressions:
    - key: node-role.kubernetes.io/control-plane
      operator: DoesNotExist
  prepare:
    args:
    - prepare
    - server-plan
    image: rancher/k3s-upgrade
  serviceAccountName: system-upgrade
  upgrade:
    image: rancher/k3s-upgrade
  version: v1.29.3-k3s1

**Comment by brandond** [comment]
There have been some recent tweaks to the SUC deployment manifest that we're still straightening out. In the mean time, you can install the CRDs yourself: `kubectl apply -f https://github.com/rancher/system-upgrade-controller/releases/download/v0.13.4/crd.yaml`

**Comment by kevinebanks** [comment]
Thanks for you response i am now getting ImagePullBackOff  on some pod for the upgrade 

![image](https://github.com/k3s-io/k3s/assets/149391336/0d9d11b8-6bb6-4748-9684-5947d0e7cb43)

**Reply by brandond** [reply]
Yes, because the `version: v1.29.3-k3s1` you've requested to upgrade to doesn't exist yet. We currently anticipate this release will finish QA early next week. Currently, only an RC is available.

**Reply by kevinebanks** [reply]
o.k. thanks
